src/renderer/components/dialogs/SetupMessage.js

```diff
@@ -41,9 +41,8 @@ const SetupMessagePartialInputSeperator = styled.div`
 class SetupMessagePanel extends React.Component {
   constructor (props) {
     super(props)
-    this.state = {
-      key: props.autocryptKey
-    }
+    this.state = { key: Array(9).fill('') }
+    this.state.key[0] = props.setupCodeBegin
```

Comment:
This child component can handle the key by itself, when the user is done it passes everything back via `onClick()` below. The parent doesn't need to know it's an array of 9 elements.